### PR TITLE
Added a retry mechanism to task.download_output()

### PR DESCRIPTION
**Pull request by Pbarbosa92**
This PR makes the download_output more reliable. 

If our task generates a large output and we have
```
task.wait()
task.download_output()
```
We can get an error telling us that there is no output (this might happen because the outputs are still being copied to the bucket). With this retry mechanism, we hope to provide enough time for the output files to be copied.

### CODE REVIEW

File: inductiva/tasks/task.py [hunk 1]
```diff
@@ -369,6 +369,9 @@ def download_outputs(
     ) -> pathlib.Path:
         """Download output files of the task.
 
+        If the API response fails for some reason, the method will retry
+        a defined amount of times before raising a RuntimeError.
```
Comment (AugustoPeres) [inline inductiva/tasks/task.py:373]: replace defined with `TASK_DOWNLOAD_MAX_API_REQUESTS`

File: inductiva/tasks/task.py [hunk 2]
```diff
@@ -379,14 +382,31 @@ def download_outputs(
...
+            except exceptions.ApiException as e:
+                failed += 1
+                if failed > constants.TASK_DOWNLOAD_MAX_API_REQUESTS:
+                    raise RuntimeError(
+                        "Something went wrong while downloading the task "
+                        "outputs. Please try again later.\n"
+                        "Keep in mind that some outputs may take some time "
+                        "to be transfered from the resource to the bucket."
+                    ) from e
+                time.sleep(1)
```
Comment (AugustoPeres) [inline inductiva/tasks/task.py:408]: maybe instead:

```python
for _ in range(constants.TASK_DOWNLOAD_MAX_API_REQUESTS):
    try:
        api_response = self._api...
        return
    time.sleep(1)
raise RuntimeError(
                        "Something went wrong while downloading the task "
                        "outputs. Please try again later.\n"
                        "Keep in mind that some outputs may take some time "
                        "to be transfered from the resource to the bucket."
                    )
```

This way we don't need `failed += 1` and `if failed > constants....` statments

**Comment by Pbarbosa92** [comment]
## QA

To test this PR we can do:
```
task= simulator.run(...)
task.kill()
task.download_output()
```

This should hit the retry and take 5 seconds to give the error message.
@AugustoPeres

**Comment by AugustoPeres** [comment]
In my case it is not waiting the expected amount of time 🔴:

Installed with:

```bash
pip install git+https://github.com/inductiva/inductiva.git@166-download_output-can-fail-if-the-output-is-big
```

Using the script:

```python
import time
import inductiva

inductiva.api_url = "https://api-dev.inductiva.ai"

input_dir = inductiva.utils.files.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "splishsplash-input-example.zip",
    unzip=True)

# Set simulation input directory
splishsplash = inductiva.simulators.SplishSplash()

task = splishsplash.run(input_dir=input_dir, sim_config_filename="config.json")

task.kill()
start_time = time.time()
task.download_outputs()
end_time = time.time()
elapsed_time = end_time - start_time
print("Elapsed time:", elapsed_time, "seconds")
```

I get

```
python qa_1347.py
Version 0.5.3 of the 'inductiva' package is available (the currently installed version is 0.5.2).
Consider upgrading to the newer version using:
  pip install --upgrade inductiva
Downloading from URL to the local path: splishsplash-input-example.zip
Uncompressing the downloaded file to: splishsplash-input-example
Task Information:
> ID:                    73r2qajnq6s85bucq4zh9j6qa
> Method:                splishsplash
> Local input directory: /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example
> Submitting to the following computational resources:
 >> Default queue with c2-standard-4 machines.
Preparing upload of the local input directory /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example (364.93 KB).
Local input directory successfully uploaded.
Task 73r2qajnq6s85bucq4zh9j6qa submitted to the default queue.
Simulation metadata logged to: inductiva_output/task_metadata.json
Task 73r2qajnq6s85bucq4zh9j6qa configurations metadata saved to the tasks metadata file task_metadata.json in the current working directory.
Consider tracking the status of the task via CLI:
	inductiva tasks list --task-id 73r2qajnq6s85bucq4zh9j6qa
Or, tracking the logs of the task via CLI:
	inductiva logs 73r2qajnq6s85bucq4zh9j6qa
Successfully sent kill request for task 73r2qajnq6s85bucq4zh9j6qa.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks list -id 73r2qajnq6s85bucq4zh9j6qa'.
Downloading simulation outputs to inductiva_output/73r2qajnq6s85bucq4zh9j6qa/output.zip.
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 768/768 [00:00<00:00, 959kB/s]
Uncompressing the outputs to inductiva_output/73r2qajnq6s85bucq4zh9j6qa.
Elapsed time: 1.5564310550689697 seconds
```

**Comment by Pbarbosa92** [comment]
> In my case it is not waiting the expected amount of time 🔴:
> 
> Installed with:
> 
> ```shell
> pip install git+https://github.com/inductiva/inductiva.git@166-download_output-can-fail-if-the-output-is-big
> ```
> 
> Using the script:
> 
> ```python
> import time
> import inductiva
> 
> inductiva.api_url = "https://api-dev.inductiva.ai"
> 
> input_dir = inductiva.utils.files.download_from_url(
>     "https://storage.googleapis.com/inductiva-api-demo-files/"
>     "splishsplash-input-example.zip",
>     unzip=True)
> 
> splishsplash = inductiva.simulators.SplishSplash()
> 
> task = splishsplash.run(input_dir=input_dir, sim_config_filename="config.json")
> 
> task.kill()
> start_time = time.time()
> task.download_outputs()
> end_time = time.time()
> elapsed_time = end_time - start_time
> print("Elapsed time:", elapsed_time, "seconds")
> ```
> 
> I get
> 
> ```
> python qa_1347.py
> Version 0.5.3 of the 'inductiva' package is available (the currently installed version is 0.5.2).
> Consider upgrading to the newer version using:
>   pip install --upgrade inductiva
> Downloading from URL to the local path: splishsplash-input-example.zip
> Uncompressing the downloaded file to: splishsplash-input-example
> Task Information:
> > ID:                    73r2qajnq6s85bucq4zh9j6qa
> > Method:                splishsplash
> > Local input directory: /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example
> > Submitting to the following computational resources:
>  >> Default queue with c2-standard-4 machines.
> Preparing upload of the local input directory /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example (364.93 KB).
> Local input directory successfully uploaded.
> Task 73r2qajnq6s85bucq4zh9j6qa submitted to the default queue.
> Simulation metadata logged to: inductiva_output/task_metadata.json
> Task 73r2qajnq6s85bucq4zh9j6qa configurations metadata saved to the tasks metadata file task_metadata.json in the current working directory.
> Consider tracking the status of the task via CLI:
> 	inductiva tasks list --task-id 73r2qajnq6s85bucq4zh9j6qa
> Or, tracking the logs of the task via CLI:
> 	inductiva logs 73r2qajnq6s85bucq4zh9j6qa
> Successfully sent kill request for task 73r2qajnq6s85bucq4zh9j6qa.
> > Consider tracking the task status by calling task.get_status()
> > or via the CLI using 'inductiva tasks list -id 73r2qajnq6s85bucq4zh9j6qa'.
> Downloading simulation outputs to inductiva_output/73r2qajnq6s85bucq4zh9j6qa/output.zip.
> 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 768/768 [00:00<00:00, 959kB/s]
> Uncompressing the outputs to inductiva_output/73r2qajnq6s85bucq4zh9j6qa.
> Elapsed time: 1.5564310550689697 seconds
> ```

Hmmmm, strange. This only happens on prod environment. 
@luispcunha @bgregoinductiva Did you guys change anything on the backend that would explain this behavior?

**Comment by bgregoinductiva** [comment]
> Hmmmm, strange. This only happens on prod environment. @luispcunha @bgregoinductiva Did you guys change anything on the backend that would explain this behavior?

The last merge to prod was on April 1st. The main difference I remember between prod and dev is that the logging stream is disabled in dev. But I didn't really understand the problem. What would you expect to happen in prod that is only happening in dev?

**Comment by Pbarbosa92** [comment]
> > Hmmmm, strange. This only happens on prod environment. @luispcunha @bgregoinductiva Did you guys change anything on the backend that would explain this behavior?
> 
> The last merge to prod was on April 1st. The main difference I remember between prod and dev is that the logging stream is disabled in dev. But I didn't really understand the problem. What would you expect to happen in prod that is only happening in dev?

In prod we get an error from the backend telling us there was no output to download. In dev there is output and we download normally. I expected both cases to generate the error off no output available. Since we kill the task right after it starts.

**Comment by Pbarbosa92** [comment]
Is no longer needed.

**Pull request closed without merging**